packages/indexer-common/src/indexing-status.ts

```diff
@@ -0,0 +1,76 @@
+import fetch from 'isomorphic-fetch'
+import gql from 'graphql-tag'
+import { Client, createClient } from '@urql/core'
+import { Logger, SubgraphDeploymentID } from '@graphprotocol/common-ts'
+import { IndexingStatus, IndexingStatusResolver } from './types'
+import { indexerError, IndexerErrorCode } from './errors'
+
+export interface IndexingStatusFetcherOptions {
+  logger: Logger
+  statusEndpoint: string
+}
+
+export class IndexingStatusFetcher implements IndexingStatusResolver {
+  logger: Logger
+  statuses: Client
+
+  constructor(options: IndexingStatusFetcherOptions) {
+    this.logger = options.logger.child({ component: 'IndexingStatusFetcher' })
+    this.statuses = createClient({
+      url: options.statusEndpoint,
+      fetch,
+      requestPolicy: 'network-only',
+    })
+  }
+
+  public async indexingStatus(deployment: SubgraphDeploymentID): Promise<IndexingStatus> {
+    try {
+      const result = await this.statuses
+        .query(
+          gql`
+            query indexingStatus($deployments: [String!]!) {
+              indexingStatuses(subgraphs: $deployments) {
+                subgraphDeployment: subgraph
+                synced
+                health
+                fatalError {
+                  handler
+                  message
+                }
+                chains {
+                  network
+                  ... on EthereumIndexingStatus {
+                    latestBlock {
+                      number
+                      hash
+                    }
+                    chainHeadBlock {
+                      number
+                      hash
+                    }
+                  }
+                }
+              }
+            }
+          `,
+          { deployments: [deployment.ipfsHash] },
+        )
+        .toPromise()
+      return (
+        result.data.indexingStatuses
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          .map((status: any) => ({
+            ...status,
+            subgraphDeployment: new SubgraphDeploymentID(status.subgraphDeployment),
+          }))
+          .pop()
```

Comment:
why pop?

Reply:
Because the `indexingStatuses` query returns an array, in this case with either no elements or a single element (the status for that deployment, if available).